### PR TITLE
Add OpenTelemetry API smoke test

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

File: test/smoke/testApps/OpenTelemetryApiSupport/build.gradle [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    archiveFileName = project.name + '.war'
+}
+
+ext.testAppArtifactDir = war.destinationDirectory
+ext.testAppArtifactFilename = project.name + '.war'
+
+dependencies {
+    compileOnly aiAgentJar // this is just to force building the agent first
+
+    compile aiWebAutoJar
```
Comment (kryalama) [inline test/smoke/testApps/OpenTelemetryApiSupport/build.gradle:20]: do we need this?

**Review by kryalama — APPROVED** [review]

File: test/smoke/testApps/OpenTelemetryApiSupport/build.gradle [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ext.testAppArtifactFilename = project.name + '.war'
+
+dependencies {
+    compileOnly aiAgentJar // this is just to force building the agent first
+
+    compile (group: 'org.springframework.boot', name: 'spring-boot-starter-web', version: '2.1.7.RELEASE') {
+        exclude group: 'org.springframework.boot', module: 'spring-boot-starter-tomcat'
+    }
+
+    compile group: 'io.opentelemetry', name: 'opentelemetry-api', version: '0.10.0'
```
Comment (heyams) [inline test/smoke/testApps/OpenTelemetryApiSupport/build.gradle:24]: do you want to test 0.11.0 based on the github wiki page you created?

File: test/smoke/testApps/OpenTelemetryApiSupport/src/main/java/com/microsoft/ajl/simple/SpringBootApp.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.microsoft.ajl.simple;
```
Comment (heyams) [inline test/smoke/testApps/OpenTelemetryApiSupport/src/main/java/com/microsoft/ajl/simple/SpringBootApp.java:1]: copyright?

File: test/smoke/testApps/OpenTelemetryApiSupport/src/main/java/com/microsoft/ajl/simple/TestController.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package com.microsoft.ajl.simple;
```
Comment (heyams) [inline test/smoke/testApps/OpenTelemetryApiSupport/src/main/java/com/microsoft/ajl/simple/TestController.java:1]: same here copyright.

File: test/smoke/testApps/OpenTelemetryApiSupport/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/OpenTelemetryApiSupportTest.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
+package com.microsoft.applicationinsights.smoketest;
```
Comment (heyams) [inline test/smoke/testApps/OpenTelemetryApiSupport/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/OpenTelemetryApiSupportTest.java:1]: same copyright

File: test/smoke/testApps/OpenTelemetryApiSupport/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/OpenTelemetryApiSupportTest.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+import java.util.List;
+
+import com.microsoft.applicationinsights.internal.schemav2.Data;
+import com.microsoft.applicationinsights.internal.schemav2.Envelope;
+import com.microsoft.applicationinsights.internal.schemav2.RemoteDependencyData;
+import com.microsoft.applicationinsights.internal.schemav2.RequestData;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
```
Comment (heyams) [inline test/smoke/testApps/OpenTelemetryApiSupport/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/OpenTelemetryApiSupportTest.java:11]: can we get rid of the wildcard? all you need are assertEquals and assertTrue.

File: test/smoke/testApps/OpenTelemetryApiSupport/build.gradle (hunk 2 above)
Comment (trask) [reply test/smoke/testApps/OpenTelemetryApiSupport/build.gradle]: good point, this branch was still on 0.10.0, I merged master into this branch and updated the test to 0.11.0

File: test/smoke/testApps/OpenTelemetryApiSupport/src/main/java/com/microsoft/ajl/simple/SpringBootApp.java (hunk 3 above)
Comment (trask) [reply test/smoke/testApps/OpenTelemetryApiSupport/src/main/java/com/microsoft/ajl/simple/SpringBootApp.java:1]: none of the smoke tests have copyright headers, if we want to change we should do across all

File: test/smoke/testApps/OpenTelemetryApiSupport/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/OpenTelemetryApiSupportTest.java (hunk 6 above)
Comment (trask) [reply test/smoke/testApps/OpenTelemetryApiSupport/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/OpenTelemetryApiSupportTest.java:11]: this is convention in all the smoke tests, if we want to change we should do across all

**Review by heyams — APPROVED** [review]

**Pull request merged**